sycl/doc/extensions/experimental/sycl_ext_oneapi_private_alloca.asciidoc

```diff
@@ -213,6 +213,35 @@ void run(queue q, const float *in, float *out, size_t n) {
   });
 ----
 
+=== Usage with `sycl::span`
+
+In this section, we show an example of how users could use this extension with
+`sycl::span` as a `std::array` replacement:
+
+[source,c++]
+----
+constexpr specialization_id<std::size_t> size(1);
+
+class Kernel;
+
+// Counterpart to 'impl' in the first example using 'sycl::span'
+SYCL_EXTERNAL void impl(const float *in, float *out,
+                        sycl::span<float> ptr);
+
+void run(queue q, const float *in, float *out, size_t n) {
+  q.submit([&](handler &h) {
+    h.set_specialization_constant<size>(n);
+    h.parallel_for<Kernel>(n, [=](id<1> i, kernel_handler kh) {
+      // Create sycl::span with the returned pointer and the specialization
+      // constant used as size.
+      sycl::span<float> tmp{
+          private_alloca<float, size, access::decorated::no>(kh),
```

Comment:
```suggestion
          private_alloca<float, size, access::decorated::no>(kh).get_raw(),
```

Don't you need to call `get_raw` here?

Reply:
I think any [iterator](https://en.cppreference.com/w/cpp/container/span/span) would work from reading this. I can add `get_raw()` anyway

Reply:
I checked: This **should** be working fine as the C++ spec defines `span` should accept any contiguous iterator there. This is however working by chance as DPC++'s headers don't define the correct constructor for `span` and we're getting a decay to pointer. I'll file an issue in DPC++ for this. Thanks, @gmlueck

Reply:
#12918 tracks this that should be addressed to conform with SYCL 2020

Reply:
My concern was that `multi_ptr` might not satisfy the constraints for the `span` constructor that takes an iterator.  According to cppreference, these constraints are:

> * It satisfies [contiguous_iterator](https://en.cppreference.com/w/cpp/iterator/contiguous_iterator),
> * the conversion from [std::iter_reference_t](http://en.cppreference.com/w/cpp/iterator/iter_t)<It> to element_type is at most a qualification conversion.

However, both of these conditions depend on C++20 features, which are not part of the SYCL 2020 specification.

Reply:
Okey, I guess I can use the raw pointer in the example to play it safe